src/main/java/pikabot/Storage.java

```diff
@@ -0,0 +1,136 @@
+package pikabot;
+
+import pikabot.task.Task;
+import pikabot.task.Deadline;
+import pikabot.task.Event;
+import pikabot.task.Todo;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Saves the tasks in the computer whenever the task list changes and
+ * loads the data from the computer when PikaBot starts up.
+ */
+public class Storage {
+
+    String filePath;
+
+    /**
+     * Constructs a Storage object.
+     *
+     * @param filePath Path of data file on computer.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Checks if data file exists on user's computer.
+     *
+     * @return Boolean value whether data file exists.
+     */
+    public boolean doesFileExist() {
+        return Files.exists(Paths.get(filePath));
+    }
+
+    /**
+     * Parses data file from computer into a TaskList.
+     *
+     * @param f Data file to be read from.
+     * @return TaskList of tasks parsed from data file.
+     * @throws FileNotFoundException If file is not found.
+     */
+    public static TaskList fileToTaskList(File f) throws FileNotFoundException {
+
+        Scanner sc = new Scanner(f);
+        ArrayList<Task> arrList = new ArrayList<>();
+        TaskList taskList = new TaskList(arrList);
+
+        while (sc.hasNextLine()) {
+            String taskStr = sc.nextLine();
+            char taskType = taskStr.charAt(1);
+            boolean isTaskDone = (taskStr.charAt(4) == 'X');
+            taskStr = taskStr.substring(7);
```

Comment:
Perhaps declaring a <code>private final String TASK_DESCRIPTION</code>  for the use of  "7" here ?